plugins/flytekit-omegaconf/flytekitplugins/omegaconf/listconfig_transformer.py

```diff
@@ -0,0 +1,124 @@
+from flytekit.models.literals import Literal, Primitive, Scalar
+from flytekit.models.types import LiteralType, SimpleType
+from omegaconf import ListConfig, OmegaConf
+
+T = TypeVar("T")
+logger = logging.getLogger("flytekit")
+
+
+class ListConfigTransformer(TypeTransformer[ListConfig]):
+    def __init__(self, mode: Optional[OmegaConfTransformerMode] = None):
+        """Construct ListConfigTransformer."""
+        super().__init__(name="OmegaConf ListConfig", t=ListConfig)
+        self.mode = mode
+
+    @property
+    def mode(self) -> OmegaConfTransformerMode:
+        """Serialisation mode for omegaconf objects defined in shared config."""
+        return SharedConfig.get_mode()
+
+    @mode.setter
+    def mode(self, new_mode: OmegaConfTransformerMode) -> None:
+        """Updates the central shared config with a new serialisation mode."""
+        SharedConfig.set_mode(new_mode)
+
+    def get_literal_type(self, t: Type[ListConfig]) -> LiteralType:
+        """
+        Provide type hint for Flytekit type system.
+
+        To support the multivariate typing of nodes in a ListConfig, we encode them as binaries (no introspection)
+        with multiple files.
+        """
+        return LiteralType(simple=SimpleType.STRUCT)
+
+    def to_literal(self, ctx: FlyteContext, python_val: T, python_type: Type[T], expected: LiteralType) -> Literal:
+        """
+        Convert from given python type object ``ListConfig`` to the Literal representation.
+
+        Since the ListConfig type does not offer additional type hints for its nodes, typing information is stored
+        within the literal itself rather than the Flyte LiteralType.
+        """
+        # instead of raising TypeError, raising AssertError so that flytekit can catch it in
+        # https://github.com/flyteorg/flytekit/blob/60c982e4b065fdb3aba0b957e506f652a2674c00/flytekit/core/
+        # type_engine.py#L1222
+        assert isinstance(python_val, ListConfig), f"Invalid type {type(python_val)}, can only serialise ListConfigs"
+
+        type_list = []
+        value_list = []
+        for idx in range(len(python_val)):
+            if OmegaConf.is_missing(python_val, idx):
+                type_list.append("MISSING")
+                value_list.append(
+                    MessageToDict(Literal(scalar=Scalar(primitive=Primitive(string_value="MISSING"))).to_flyte_idl())
+                )
+            else:
+                node_type, type_name = extract_node_type(python_val, idx)
+                type_list.append(type_name)
+
+                transformation_logs = ""
+                transformer = TypeEngine.get_transformer(node_type)
+                literal_type = transformer.get_literal_type(node_type)
+                value_list.append(
+                    MessageToDict(transformer.to_literal(ctx, python_val[idx], node_type, literal_type).to_flyte_idl())
+                )
+
+                if len(type_list) != len(value_list):
```

Comment:
this check is always false, no?

Reply:
addressed in https://github.com/flyteorg/flytekit/pull/2299/commits/709d4a4364e94f2e531f533c076b564dc3a26d38